src/elisp/treemacs-core-utils.el

```diff
@@ -1272,11 +1273,33 @@ GOTO-TAG: Bool"
 
 (defun treemacs-is-path-visible? (path)
   "Return whether a node for PATH is displayed in the current buffer.
-The return value, if PATH is visible, is either the dom node of PATH - if it
-is an expanded directory - or the dom node of its parent - if it is a dir or
-file below an expanded directory."
-  (or (treemacs-find-in-dom path)
-      (treemacs-find-in-dom (treemacs--parent path))))
+The return value is a boolean indicating whether the node is visible."
+  (let* ((current-node (treemacs-find-in-dom path))
+         (current-parent (treemacs-find-in-dom (treemacs--parent path))))
```

Comment:
That won't work here, since `current-parent` is needed in the case where the child being queried is a leaf node that does not exist in DOM, so it's possible that `current-node` is `nil` and `current-parent` is a visible expanded DOM node.